libs/pascal_voc_io.py

```diff
@@ -130,15 +150,44 @@ def __init__(self, filepath):
         # shapes type:
         # [labbel, [(x1,y1), (x2,y2), (x3,y3), (x4,y4)], color, color, difficult]
         self.shapes = []
+        self.components = []
         self.filepath = filepath
         self.verified = False
         try:
             self.parseXML()
         except:
             pass
+    
+    def sortShapes(self):
+        heights = []
+        for i in self.shapes:
+            heights.append(i[1][2][1]-i[1][0][1])
+
+        maxHeight = max(heights)
+        resList = copy.deepcopy(self.components)
+        t = 0
```

Comment:
What is the purpose of `t`  variable here?
I think it will be a good idea to have the option to allow users to change the sort method. For example, by default, sort by adding order, but they can change the order method somewhere in menu

Reply:
I recently made a lot of annotations in similary size, such as words in the images. In order to sort these annotations, my solution is set a variable  `t`  to find out words in a row first . For example , if the annotation starts with left-top point (0,0) and I want to filter those words between [0,100] in the height , I need to set a regular '100' or a rate `t` , then I sort the list of objects from left to right. 
I used the maximum height of all the annotated object as the default regular. 
Besides, I think get the first annotation ( the annotation most close to (0,0) ), its (xmin,ymin) point to determine `t` is much better, which could be :
`t = ymin/image.shape[0]`